### PR TITLE
filesystem: add bcachefs support

### PR DESCRIPTION
**Pull request by stintel**
##### SUMMARY

Add support for bcachefs to the filesystem module.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

filesystem

##### ADDITIONAL INFORMATION

Test playbook:
```yaml
---
- hosts: localhost
  vars:
    img_file: /tmp/bcachefs.img
  tasks:
    - name: create 1GiB file
      community.general.filesize:
        path: "{{ img_file }}"
        size: 1GiB

    - name: create bcachefs filesystem on file
      ansible.legacy.filesystem:
        dev: "{{ img_file }}"
        fstype: bcachefs
        uuid: bfe441c5-93da-440b-8c9a-e807bc538abb

    - name: create bcachefs filesystem on file (no changes)
      ansible.legacy.filesystem:
        dev: "{{ img_file }}"
        force: False
        fstype: bcachefs

    - name: create bcachefs filesystem on file (force)
      ansible.legacy.filesystem:
        dev: "{{ img_file }}"
        force: True
        fstype: bcachefs

    - name: resize 1GiB file to 2GiB
      community.general.filesize:
        path: "{{ img_file }}"
        size: 2GiB

    - name: resize bcachefs filesystem on file
      ansible.legacy.filesystem:
        dev: "{{ img_file }}"
        fstype: bcachefs
        resizefs: True
```
Output:
```
PLAY [localhost] *****************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************
ok: [localhost]

TASK [create 1GiB file] **********************************************************************************************************************************
--- before
+++ after
@@ -1 +1 @@
-size: null
+size: 1073741824

changed: [localhost]

TASK [create bcachefs filesystem on file] ****************************************************************************************************************
changed: [localhost]

TASK [create bcachefs filesystem on file (no changes)] ***************************************************************************************************
ok: [localhost]

TASK [create bcachefs filesystem on file (force)] ********************************************************************************************************
changed: [localhost]

TASK [resize 1GiB file to 2GiB] **************************************************************************************************************************
--- before
+++ after
@@ -1 +1 @@
-size: 1073741824
+size: 2147483648

changed: [localhost]

TASK [resize bcachefs filesystem on file] ****************************************************************************************************************
changed: [localhost]

PLAY RECAP ***********************************************************************************************************************************************
localhost                  : ok=7    changed=5    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

### CODE REVIEW

**Comment by stintel** [comment]
The UUID checks are most likely failing because the util-linux version on those distros does not have support for bcachefs. I'll try to look at that over the weekend.

File: changelogs/fragments/8126-filesystem-bcachefs-support.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - add bcachefs support
```
Comment (felixfontein) [inline changelogs/fragments/8126-filesystem-bcachefs-support.yaml:2]:
```suggestion
  - filesystem - add bcachefs support (https://github.com/ansible-collections/community.general/pull/8126).
```

File: plugins/modules/filesystem.py [hunk 2]
```diff
@@ -40,7 +40,7 @@
     default: present
     version_added: 1.3.0
   fstype:
-    choices: [ btrfs, ext2, ext3, ext4, ext4dev, f2fs, lvm, ocfs2, reiserfs, xfs, vfat, swap, ufs ]
+    choices: [ bcachefs, btrfs, ext2, ext3, ext4, ext4dev, f2fs, lvm, ocfs2, reiserfs, xfs, vfat, swap, ufs ]
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:43]: Can you add another entry to the `description` below: `bcachefs support has been added in community.general 8.5.0.`?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments below.

**Comment by stintel** [comment]
Would it be fine to limit the tests to Alpine >= 3.19 for now? That seems to be the only distribution with recent enough versions of both bcachefs-tools and util-linux used in the test matrix.

**Comment by felixfontein** [comment]
I would expect that at least with Arch Linux the tests should also pass, since Arch usually has very up to date packages. (https://archlinux.org/packages/core/x86_64/util-linux/, https://archlinux.org/packages/extra/x86_64/bcachefs-tools/)

**Comment by stintel** [comment]
You're right, looking at the versions I'd expect it to work. I modified the tests to run bcachefs related tasks only on Alpine and Arch Linux.

File: tests/integration/targets/filesystem/tasks/main.yml [hunk 3]
```diff
@@ -46,6 +46,10 @@
 
     # Other limitations and corner cases
 
+    # bcachefs only on Alpine > 3.18 and Arch Linux for now
+    # other distributions have too old versions of bcachefs-tools and/or util-linux (blkid for UUID tests)
+    - 'ansible_distribution == "Alpine" and ansible_distribution_version is version("3.18", ">") and item.0.key == "bcachefs"'
+    - 'ansible_distribution == "Archlinux and item.0.key == "bcachefs"'
```
Comment (felixfontein) [inline tests/integration/targets/filesystem/tasks/main.yml:52]:
```suggestion
    - 'ansible_distribution == "Archlinux" and item.0.key == "bcachefs"'
```

Comment (stintel) [reply tests/integration/targets/filesystem/tasks/main.yml]: Sorry about that, I should have just gone to bed and look at it today.

Comment (felixfontein) [reply tests/integration/targets/filesystem/tasks/main.yml]: No worries :)

File: plugins/modules/filesystem.py [hunk 4]
```diff
@@ -40,11 +40,12 @@
...
     description:
       - Filesystem type to be created. This option is required with
         O(state=present) (or if O(state) is omitted).
       - ufs support has been added in community.general 3.4.0.
+      - bcachefs support has been added in community.general 8.5.0.
```
Comment (felixfontein) [inline plugins/modules/filesystem.py:48]:
```suggestion
      - bcachefs support has been added in community.general 8.6.0.
```

**Comment by felixfontein** [comment]
@stintel thanks for your contribution!
